plugin.js

```diff
@@ -39,10 +39,10 @@ function plugin (fastify, options, pluginRegistrationDone) {
     )
   }
 
-  fastify.decorateRequest('session', getSession())
+  fastify.decorateRequest('session', null)
   fastify.addHook('onRequest', function (req, reply, hookFinished) {
+    req.session = getSession()
     if (!req.cookies[opts.sessionCookieName]) {
-      req.session = getSession()
       return hookFinished()
     }
```

Comment:
This is going to create a new session on every incoming request. Is it not?

Reply:
This is worth adding test assertions. I'll work on it.

Reply:
I guess this is not creating a new session on every request if this test passes: https://github.com/jsumners/fastify-server-session/blob/39340474a20d67bafbdf06fa3a3e95579800002f/test/session.test.js#L260-L304